tests/datasource/fluent/integration/conftest.py

```diff
@@ -136,8 +143,12 @@ def sql_data(
 
 
 def spark_filesystem_datasource(
+    test_backends,
     context: AbstractDataContext,
 ) -> SparkFilesystemDatasource:
+    if "SparkDFDataset" not in test_backends:
+        pytest.skip("No spark backend selected.")
```

Comment:
Moving Spark connection logic into `test_connection` for the datasources meant that we needed to skip the tests earlier when spark wasn't installed.